notificationservice/src/main/java/com/example/notificationservice/controllers/SmsController.java

```diff
@@ -0,0 +1,98 @@
+package com.example.notificationservice.controllers;
+
+import  com.example.notificationservice.models.responsebody.SmsResponseBody;
+import  com.example.notificationservice.models.requestbody.SmsRequestBody;
+import com.example.notificationservice.services.sms.SmsService;
+import  com.example.notificationservice.models.SMS;
+import  com.example.notificationservice.repositories.SmsRepository;
+import com.example.notificationservice.services.blacklist.BlacklistService;
+import com.example.notificationservice.models.requestbody.BlacklistRequestBody;
+import com.example.notificationservice.models.responsebody.BlacklistResponseBody;
+
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import lombok.RequiredArgsConstructor;
+
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1")
+@RequiredArgsConstructor(onConstructor_=@Autowired)
+public class SmsController {
+
+    private final SmsService smsService;
+    private final SmsRepository smsRepository;
+    private final BlacklistService blacklistService;
+
+    private static final Logger LOGGER = LogManager.getLogger(SmsController.class);
+
+    @GetMapping
+    public List<SMS> getAllSMS(){
+        return smsRepository.findAll();
+    }
+
+    @PostMapping
+    public SMS createSMS(@RequestBody SMS sms) {
+        return smsRepository.save(sms);
```

Comment:
We shouldn't directly accept the payload as an entity object. Create a model class which accepts the payload and construct the Entity object with this info. Also, create a DTO (Data transfer object) whose structure will be similar to entity object and construct this object with the help of persisted entity object. This DTO object is ultimately returned back to controller, not the entity object.